### PR TITLE
feat(NewPost validations): chapter 2

### PR DESCRIPTION
**Pull request by JetoPistola**
https://drive.google.com/open?id=1QYz24gnUK8-E2I5zkDNte3UmccXv2V9p

Link to the video 😃 , you have it in `Teams` too.

What I did in this merge, in short:

1. I refactored `HebrewDatePicker` (Style, tooltips for error case, validations and etc)
2. I refactored `Switch` to be like in PPTX,
3. I added `locationBase` to input
4. I refactored the `whole modal` to the updated PPTX (Logical locations)
5. I added `Colors` and `Fonts` types, so we will have less `MAGIC` code and more `const` code

Fee free to comment!!!

### CODE REVIEW

File: src/assets/projectJSS/Colors.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// File contains all theme colors through the project, documented in 1st page of PowerPoint/Pdf
+
+// Usage: color: `${NAV_BAR_COLOR}`,
+
+export const NAV_BAR_COLOR = 'rgb(40,54,67)';
```
Comment (SteveVaknin) [inline src/assets/projectJSS/Colors.ts:5]: I think thats what `theme` is for.   
You make global attributes that are provided by `materia-ui` to all the components.

File: src/assets/projectJSS/Colors.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+export const DARK_TEST_COLOR = '#595959'; // For advertisments mostly
+export const AVATAR_COLOR = 'rgb(73,90,114)';
+export const TOOLTIP_OR_SCROLLER_COLOR = 'rgb(127, 140, 156)';
+export const TEXT_AND_ICONS_ADS_COLOR = 'rgb(46, 188, 145)'; // Advertisements
+export const TEXT_AND_ICONS_ADS_BACKGROUND_COLOR = '#F8F9FC'; // Advertisements
+export const NEW_JOB_COLOR = '#21BD90';
+export const YELLOW_STAR_COLOR = 'rgb(254,201,62)'; // Same as in PowerPoint
+
+// Shadow similar to PowerPoint
+export const BOX_SHADOW = "0px 1px 7px 1px rgba(0,0,0,0.75)";
```
Comment (SteveVaknin) [inline src/assets/projectJSS/Colors.ts:16]: IMO it's better to have a jss like object that you can spread.  
For example:
```jsx
const lightShadow = {
    boxShadow: '0px 1px 7px 1px rgba(0,0,0,0.75)'
};

// ... the component's jss class
{
adContainer: {
   ...lightBoxShadow
},
```

File: src/components/Application-Body/JobsAppBody.tsx [hunk 3]
```diff
@@ -5,38 +5,44 @@ import Button from '@material-ui/core/Button';
...
     appBodyContent: {
-        paddingLeft: '2vw',
+        paddingLeft: '3vw',
         paddingRight: '2vw',
         paddingTop: '2vh',
     },
     addNewPostButton: {
-        color: "white",
-        backgroundColor: `${dialogThemeColor}`,
+        color: `${LIGHT_COLOR_TEXT}`,
```
Comment (SteveVaknin) [inline src/components/Application-Body/JobsAppBody.tsx:18]: Aren't they `string` already?

File: src/components/Application-Body/PostNewJob/JobEntryDateInput.tsx [hunk 4]
```diff
@@ -1,52 +1,106 @@
...
+import { KeyboardDatePicker } from "@material-ui/pickers/DatePicker/DatePicker";
 import { MaterialUiPickersDate } from '@material-ui/pickers/typings/date';
 
 import styles from './PostNewJobStyles';
 import SwitchInput from './SwitchInput';
+import format from 'date-fns/format';
+import startOfMonth from 'date-fns/startOfMonth';
+import endOfMonth from 'date-fns/endOfMonth';
+import addYears from 'date-fns/addYears';
+import { ParsableDate } from '@material-ui/pickers/constants/prop-types';
```
Comment (SteveVaknin) [inline src/components/Application-Body/PostNewJob/JobEntryDateInput.tsx:18]: Should be in the import group above (of materia-ui)

File: src/components/Application-Body/PostNewJob/JobNicknameInput.tsx [hunk 5]
```diff
@@ -1,38 +1,82 @@
...
-import Typography from '@material-ui/core/Typography';
+import InputLabel from '@material-ui/core/InputLabel';
 import TextField from '@material-ui/core/TextField';
 
 import styles from './PostNewJobStyles';
 
 interface JobNicknameInputProps {
     jobNickname: string;
     setJobNickname: (jobNickname: string) => void;
+    didValidationFail: boolean;
```
Comment (SteveVaknin) [inline src/components/Application-Body/PostNewJob/JobNicknameInput.tsx:12]: Perhapse `isValid` instead?

File: src/assets/projectJSS/Colors.ts (hunk 1 above)
Comment (JetoPistola) [reply src/assets/projectJSS/Colors.ts:5]: Yes, but the `theme` is not global for `all cases` (Not all buttons in project are green).

I left the all colors in 1 place for comfortable. If such color does repeat... then it is added inside the global theme 😃

File: src/components/Application-Body/JobsAppBody.tsx (hunk 3 above)
Comment (JetoPistola) [reply src/components/Application-Body/JobsAppBody.tsx:18]: You are correct! thanks 😃

File: src/assets/projectJSS/Colors.ts (hunk 2 above)
Comment (JetoPistola) [reply src/assets/projectJSS/Colors.ts:16]: Done 😄

File: src/components/Application-Body/PostNewJob/JobEntryDateInput.tsx (hunk 4 above)
Comment (JetoPistola) [reply src/components/Application-Body/PostNewJob/JobEntryDateInput.tsx:18]: Done 😃

**Pull request merged**